### ISSUE TITLE
How to get access token with out popup login page ?

### ISSUE THREAD

**Issue by bluedusk** [microsoftgraph/microsoft-graph-docs#3162]
Hi,
What I'm doing is a server end app that read & write to a user's mailbox.
When using graph I need to get access key. Question is how to get the token without requiring user input password on popup login form ?? Thanks

**Comment by bluedusk** [comment]
Is there a way to aquire token without user to input username/password on page and click login ???

**Comment by damien122** [comment]
I think your way is not possible, because you should give the user a possibility to define email and password to login to backend and allow your app scope.

**Comment by bluedusk** [comment]
@damien122  Say I have a username/password. I suppose there might be a grant flow enables backend app to fetch access token without user presence. (e.g. grant_type = password).
I tried but no luck.

**Comment by damien122** [comment]
@bluedusk you can try this ->[ https://developer.microsoft.com/en-us/graph/docs/concepts/auth_v2_user]( https://developer.microsoft.com/en-us/graph/docs/concepts/auth_v2_user) and an implicit grant -> [https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-implicit-grant-flow](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-implicit-grant-flow)

I use the Implicit grant in my app cloudnoter, which you can find in my repos. Its an Javascript application. 
But there is no possible way to get the access without user interaction. Because if the user start your app for the first time, they should approve that your app can access the user account with the grant you define.

**Comment by bluedusk** [comment]
@damien122 Have you tried OAuth grant flow with grant_type = password ?

**Comment by damien122** [comment]
@bluedusk nope. In my case I need the bearer token to access onenote stuff.

**Comment by damien122** [comment]
@bluedusk I use the implicit grant type and the response is a bearer token. You should send the token in the header at each request. You can have a look here [My Code](https://github.com/damien122/cloudnoter/blob/fix-login/index.js) see at line 192. This is the authentication and later also a callback is defined which handle the response.

**Comment by damien122** [comment]
@bluedusk is your porblem fixed now? was it possible for you to login?
